scripts/dbgen/main.go

```diff
@@ -160,6 +165,78 @@ const (
 	return os.WriteFile(outputPath, data, 0o600)
 }
 
+// generateForeignKeyConstraints generates the ForeignKeyConstraint enum.
+func generateForeignKeyConstraints() error {
+	localPath, err := localFilePath()
+	if err != nil {
+		return err
+	}
+	databasePath := filepath.Join(localPath, "..", "..", "..", "coderd", "database")
+
+	dump, err := os.Open(filepath.Join(databasePath, "dump.sql"))
+	if err != nil {
+		return err
+	}
+	defer dump.Close()
+
+	var foreignKeyConstraints []string
+	dumpScanner := bufio.NewScanner(dump)
+	query := ""
+	for dumpScanner.Scan() {
+		line := strings.TrimSpace(dumpScanner.Text())
+		switch {
+		case strings.HasPrefix(line, "--"):
+		case line == "":
+		case strings.HasSuffix(line, ";"):
+			query += line
+			if strings.Contains(query, "FOREIGN KEY") {
+				foreignKeyConstraints = append(foreignKeyConstraints, query)
+			}
+			query = ""
+		default:
+			query += line + " "
+		}
+	}
+
+	if err := dumpScanner.Err(); err != nil {
+		return err
+	}
+
+	s := &bytes.Buffer{}
+
+	_, _ = fmt.Fprint(s, `// Code generated by gen/enum. DO NOT EDIT.
+package database
+`)
+	_, _ = fmt.Fprint(s, `
+// ForeignKeyConstraint represents a named foreign key constraint on a table.
+type ForeignKeyConstraint string
+
+// ForeignKeyConstraint enums.
+const (
+`)
+	for _, query := range foreignKeyConstraints {
+		name := ""
+		switch {
+		case strings.Contains(query, "ALTER TABLE") && strings.Contains(query, "ADD CONSTRAINT"):
+			name = strings.Split(query, " ")[6]
```

Comment:
parsing is pretty flimsy, but I guess the rest of the file is no better...

Reply:
Yeah... it works for the moment.

Reply:
I don't think it's worth improving, SQL is pretty stable 😄.